pilot/pkg/networking/core/v1alpha3/tls.go

```diff
@@ -247,10 +235,11 @@ func buildSidecarOutboundTCPFilterChainOpts(node *model.Proxy, env *model.Enviro
 	return out
 }
 
-func buildSidecarOutboundTCPTLSFilterChainOpts(node *model.Proxy, env *model.Environment, configs []model.Config, destinationIPAddress string,
-	service *model.Service, listenPort *model.Port, proxyLabels model.LabelsCollection, gateways map[string]bool) []*filterChainOpts {
+func buildSidecarOutboundTCPTLSFilterChainOpts(node *model.Proxy, env *model.Environment, destinationIPAddress string,
+	service *model.Service, listenPort *model.Port, proxyLabels model.LabelsCollection, gateways map[string]bool,
+	byHost virtualServiceForHost) []*filterChainOpts {
 
-	virtualService := getVirtualServiceForHost(service.Hostname, configs)
+	virtualService := byHost(service.Hostname)
```

Comment:
Can we keep this in the simple function call way and not pass functions as parameters and have implicit variables in the functions? Its such a pain to follow the code.